client/packages/invoices/src/InboundShipment/DetailView/Toolbar.tsx

```diff
@@ -49,6 +53,14 @@ export const Toolbar: FC = () => {
   ]);
   const { isGrouped, toggleIsGrouped } = useInbound.lines.rows();
   const t = useTranslation('replenishment');
+
+  const generateNewReturnLines = useInbound.lines.newReturnLines();
+
+  const onReturn = async () => {
+    const lines = await generateNewReturnLines();
+    if (lines) onReturnLines(lines);
+  };
```

Comment:
I'm not sure how I feel about giving the Toolbar the responsibility of querying for the new return lines, and only opening the modal if there are any.

Part of me wanted to do the querying for new return lines within the modal, but I don't like the idea of rendering the modal, only to determine there are no lines selected and having to close itself again?